networkx/algorithms/dag.py

```diff
@@ -287,3 +289,83 @@ def is_aperiodic(G):
         return g==1
     else:
         return g==1 and nx.is_aperiodic(G.subgraph(set(G)-set(levels)))
+
+def transitive_closure(G):
+    """ Returns transitive closure of a directed graph
+
+    The transitive closure of G = (V,E) is a graph G+ = (V,E+) such that
+    for all v,w in V there is an edge (v,w) in E+ if and only if there
+    is a non-null path from v to w in G.
+
+    Parameters
+    ----------
+    G : NetworkX DiGraph
+      Graph
+
+    Returns
+    -------
+    TC : NetworkX DiGraph
+       Graph
+
+    Raises
+    ------
+    NetworkXError
+      If G is not directed
+
+    References
+    ----------
+    .. [1] http://www.ics.uci.edu/~eppstein/PADS/PartialOrder.py
+
+    """
+    if not G.is_directed():
+        raise nx.NetworkXError("transitive_closure not defined for undirected graphs")
+    TC = nx.DiGraph()
+    TC.add_nodes_from(G.nodes_iter())
+    TC.add_edges_from(G.edges_iter())
+    for v in G:
+        TC.add_edges_from((v, u) for u in nx.dfs_preorder_nodes(G, source=v)
+                          if v != u)
+    return TC
+
+def antichains(G):
+    """Generates antichains from a DAG.
+    An antichain is a subset of a partially ordered set such that any
+    two elements in the subset are incomparable.
+
+    Parameters
+    ----------
+    G : NetworkX DiGraph
+      Graph
+
+    Returns
+    -------
+    antichain : generator object
+
+    Raises
+    ------
+    NetworkXError
+      If G is not directed
+
+    Notes
+    -----
+    This function was originally developed by Peter Jipsen and Franco Saliola
+    for the SAGE project. It's included in NetworkX with permission from the
+    authors. Original SAGE code at:
+
+    https://sage.informatik.uni-goettingen.de/src/combinat/posets/hasse_diagram.py
+
```

Comment:
Could you add a `References` section with this:
Free Lattices, by R. Freese, J. Jezek and J. B. Nation, AMS, Vol 42, 1995, p. 226.